core/src/main/java/org/bitcoinj/uri/BitcoinURI.java

```diff
@@ -147,6 +149,11 @@ public BitcoinURI(String input, @Nonnull Network network) throws BitcoinURIParse
             throw new BitcoinURIParseException("Bad URI syntax", e);
         }
 
+        // TODO: Determine actual network by parsing the address field of the input URI
+        // TODO: Discover (via Service Loader mechanism) alt-Networks from the scheme/prefix in the uri string
+        Network network = autoNetwork ? BitcoinNetwork.MAINNET : nullableNetwork;
```

Comment:
Isn't that still hardcoded to mainnet?

The previous behaviour was that it would parse either mainnet or testnet URIs (and I hoped in future also signet), and the app presents a specific warning to the user if there is a mismatch between network (which happens frequently!).

We need basically the same behaviour as `AddressParser.parseAddressAnyNetwork()`.

Reply:
So I think I it needs two conditional branches for line 200: one for if the network is given, and one for if it isn't.

Reply:
> Isn't that still hardcoded to mainnet?

Yes. It doesn't change behavior from the current `master`. This PR just restructures constructors for the change you are looking for.